js/reference.js

```diff
@@ -2723,12 +2721,24 @@ var ERMrest = (function(module) {
                         if (column.isPseudo) {
                             if (column.isForeignKey) {
                                 presentation = column.formatPresentation(this._linkedData[column._constraintName], {context: this._pageRef._context});
-                            } else {
+                            }
+                            else {
                                 presentation = column.formatPresentation(this._data, { formattedValues: keyValues, context: this._pageRef._context});
                             }
                             this._values[i] = presentation.value;
                             this._isHTML[i] = presentation.isHTML;
-                        } else {
+                        }
+                        else if(column.type.name === ("json" || "jsonb")){
+                            presentation = column.formatPresentation(keyValues[column.name], { formattedValues: keyValues , context: this._pageRef._context });
+                            if(typeof keyValues['_'+column.name] === "string"){
```

Comment:
The purpose of the if statement, where it checks for JSON and jsonb, is to pass the formatted value (string representation) of the json.